modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/OrderSummaryCommerceCheckoutStep.java

```diff
@@ -318,6 +320,10 @@ private void _validateCommerceOrder(
 			_commerceOrderService.getCommerceOrderByUuidAndGroupId(
 				commerceOrderUuid, groupId);
 
+		if (!_commerceShippingHelper.isAvailable(commerceOrder)) {
```

Comment:
- I don't think you should add this method to the shippingHelper
- This method should check the visibilty not the availability, on this behalf we should also check if the product is visible also for the account
- I don't think this is the right name for the exception, the CommerceChannelRel is not required

Reply:
Also I don't get why you check this in the Summary checkout step, this means that you first create the order with item, I would avoid to add this item to the new order if it cannot be added, so I would add this validation to the order validation/order item validation

Reply:
@andrea-ale-sbarra the orderItem is still created and that's fine. This way we can show the error to the user.
But I agree that this is not the proper solution.
@victorcdp you should create a new implementation of `CommerceOrderValidator` or at least adding the logic to `DefaultCommerceOrderValidatorImpl`, so that the error can be properly managed.